DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorRecHitsSoAGPU.cc

```diff
@@ -0,0 +1,116 @@
+// -*- C++ -*-
+///bookLayer
+// Package:    SiPixelPhase1MonitorRecHitsSoAGPU
+// Class:      SiPixelPhase1MonitorRecHitsSoAGPU
+//
+/**\class SiPixelPhase1MonitorRecHitsSoAGPU SiPixelPhase1MonitorRecHitsSoAGPU.cc 
+*/
+//
+// Author: Alessandro Rossi, Suvankar Roy Chowdhury
+//
+#include "HeterogeneousCore/CUDACore/interface/ScopedContext.h"
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DHeterogeneous.h"
+// for string manipulations
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "SiPixelPhase1MonitorRecHitsSoABase.h"
+
+class SiPixelPhase1MonitorRecHitsSoAGPU : public SiPixelPhase1MonitorRecHitsSoABase {
+public:
+  explicit SiPixelPhase1MonitorRecHitsSoAGPU(const edm::ParameterSet&);
+  ~SiPixelPhase1MonitorRecHitsSoAGPU() override = default;
+  void analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  edm::EDGetTokenT<cms::cuda::Product<TrackingRecHit2DGPU>> tokenHitsGPU_;
+  edm::EDGetTokenT<SiPixelClusterCollectionNew> clusterToken_;
+  cms::cuda::host::unique_ptr<float[]> m_store32;
+  cms::cuda::host::unique_ptr<uint32_t[]> m_hitsModuleStart;
```

Comment:
These would be better as local variables (because they are used in only one member function).